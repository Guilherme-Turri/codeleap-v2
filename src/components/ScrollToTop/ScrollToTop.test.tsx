import { ScrollToTop } from '.'
import { componentRender } from '../../render/componentRender'
import { screen, fireEvent } from '@testing-library/react'

describe('<ScrollToTop/>', () => {
  it('Should render on screen and call function', () => {
    window.scrollTo = jest.fn()
    const { container } = componentRender(<ScrollToTop />)
    const scroll = screen.getByTestId('scroll')
    expect(scroll).toBeInTheDocument();
    fireEvent.click(scroll);
    expect(window.scrollTo).toHaveBeenCalledWith(0, 0)
    expect(container).toMatchSnapshot();
  })
})