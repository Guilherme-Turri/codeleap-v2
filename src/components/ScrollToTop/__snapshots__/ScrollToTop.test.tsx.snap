// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollToTop/> Should render on screen and call function 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  color: #00CBA9;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  width: 50px;
  height: 50px;
  bottom: 0.938rem;
  right: 15px;
  cursor: pointer;
  z-index: 999999;
}

<div>
  <a
    class="c0"
    data-testid="scroll"
  >
    <div
      class="c1"
      data-testid="UpArrow"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 14h10l-5-6z"
        />
        <path
          d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
        />
      </svg>
    </div>
  </a>
</div>
`;
